src/main/java/io/jenkins/plugins/synopsys/security/scan/extension/pipeline/SecurityScanStep.java

```diff
@@ -219,7 +224,13 @@ public void setBlackduck_install_directory(String blackduck_install_directory) {
 
     @DataBoundSetter
     public void setBlackduck_scan_full(Boolean blackduck_scan_full) {
-        this.blackduck_scan_full = blackduck_scan_full;
+        if (blackduck_scan_full) {
+            this.blackduckIntelligentScan = true;
+        }
+        if (!blackduck_scan_full) {
+            this.blackduckIntelligentScan = false;
```

Comment:
this logic can be shortened like this `this.blackduckIntelligentScan = blackduck_scan_full != null && blackduck_scan_full;`